src/reference/asciidoc/kafka.adoc

```diff
@@ -380,6 +380,75 @@ public void listen(@Payload String foo,
 }
 ----
 
+Starting with _version 1.1_, `@KafkaListener` methods can be configured to receive the entire batch of consumer records received from the consumer poll.
+To configure the listener container factory to create batch listeners, set the `batchListener` property:
+
+[source, java]
+----
+@Bean
+public KafkaListenerContainerFactory<?> batchFactory() {
+    ConcurrentKafkaListenerContainerFactory<Integer, String> factory =
+            new ConcurrentKafkaListenerContainerFactory<>();
+    factory.setConsumerFactory(consumerFactory());
+    factory.setBatchListener(true);  // <<<<<<<<<<<<<<<<<<<<<<<<<
```

Comment:
I see that it isn't possible (yet) to determine batching or record-based target listener at runtime, although all that `MessagingMessageListenerAdapter.determineInferredType()` says opposite and looks like we can somehow find a combination of both `BatchMessagingMessageListenerAdapter` and `RecordMessagingMessageListenerAdapter`.

But let's, at least, add `batching = true/false` logic into the `@KafkaListener` annootation!
WDYT?